src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,490 @@
+                return;
+            }
+
+            // The ScriptHost is already monitoring for changes to function.json, so we skip those
+            string fileExtension = Path.GetExtension(e.Name);
+            if (WatchedFileTypes.Contains(fileExtension))
+            {
+                _reloadScript();
+            }
+        }
+        // private void InvokePowerShellScript()
+        private void InvokePowerShellScript()
+        {
+            InitialSessionState iss = InitialSessionState.CreateDefault();
+            _moduleFiles = GetModuleFilePaths();
+
+            using (Runspace runspace = RunspaceFactory.CreateRunspace(iss))
+            {
+                runspace.Open();
+                RunspaceInvoke runSpaceInvoker = new RunspaceInvoke(runspace);
+                runSpaceInvoker.Invoke("Set-ExecutionPolicy -Scope Process -ExecutionPolicy Unrestricted");
+
+                using (
+                    System.Management.Automation.PowerShell powerShellInstance =
+                        System.Management.Automation.PowerShell.Create())
+                {
+                    powerShellInstance.Runspace = runspace;
+                    _moduleFiles = GetModuleFilePaths();
+                    if (_moduleFiles.Any())
+                    {
+                        powerShellInstance.AddCommand("Import-Module").AddArgument(_moduleFiles);
+                        LogLoadedModules();
+                    }
+
+                    _script = GetScript();
+                    powerShellInstance.AddScript(_script, true);
+
+                    PSDataCollection<PSObject> outputCollection = new PSDataCollection<PSObject>();
+                    outputCollection.DataAdded += OutputCollectionDataAdded;
+
+                    powerShellInstance.Streams.Error.DataAdded += ErrorDataAdded;
+
+                    IAsyncResult result = powerShellInstance.BeginInvoke<PSObject, PSObject>(null, outputCollection);
+
+                    while (result.IsCompleted == false)
+                    {
+                        System.Threading.Thread.Sleep(50);
+                    }
+
+                    // Clean up and remove module
+                    if (_moduleFiles.Any())
+                    {
+                        powerShellInstance.AddCommand("Remove-Module").AddArgument(_moduleFiles);
+                    }
+                }
+
+                runspace.Close();
+            }
+        }
+
+        private void LogLoadedModules()
+        {
+            List<string> moduleRelativePaths = new List<string>();
+            foreach (string moduleFile in _moduleFiles)
+            {
+                string relativePath = GetRelativePath(moduleFile);
+                moduleRelativePaths.Add(relativePath);
+            }
+
+            if (moduleRelativePaths.Any())
+            {
+                TraceWriter.Info(string.Format("Loaded modules:{0}{1}", Environment.NewLine, string.Join(Environment.NewLine, moduleRelativePaths)));
```

Comment:
Wondering if this should be Info or Verbose? Will people want to see this always? Or only opt in, if the set the level to Verbose?